src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1396,6 +1396,12 @@ private uint getClassSize(CORINFO_CLASS_STRUCT_* cls)
             {
                 throw new RequiresRuntimeJitException(type);
             }
+
+            if (NeedsTypeLayoutCheck(type))
+            {
+                ISymbolNode node = _compilation.SymbolNodeFactory.CheckTypeLayout(type);
+                ((MethodWithGCInfo)_methodCodeNode).Fixups.Add(node);
```

Comment:
Unrelated to your change but something we might want to simplify in a follow-up change - IReadyToRunMethodCodeNode seems somewhat superfluous as MethodWithGCInfo is its only implementor in crossgen2. @MichalStrehovsky - could you comment on this please?

Reply:
Yup, it doesn't look like the interface brings anything but trouble. The full AOT compiler had a similar interface here to facilitate code sharing between the compiler and a JIT prototype, but we got rid of that interface in the meantime because it wasn't that useful either.

Reply:
Good idea; I can do that in a follow-up PR